src/decode.rs

```diff
@@ -2555,58 +2552,68 @@ unsafe fn decode_b_inner(
             }
 
             let im = &dav1d_comp_inter_pred_modes[b.inter_mode() as usize];
-            *b.drl_idx_mut() = NEAREST_DRL;
+            *b.drl_idx_mut() = DrlProximity::Nearest;
             if b.inter_mode() == NEWMV_NEWMV {
                 if n_mvs > 1 {
                     // NEARER, NEAR or NEARISH
                     let drl_ctx_v1 = get_drl_context(&mvstack, 0);
-                    *b.drl_idx_mut() += rav1d_msac_decode_bool_adapt(
+                    if rav1d_msac_decode_bool_adapt(
                         &mut ts.msac,
                         &mut ts.cdf.m.drl_bit[drl_ctx_v1 as usize],
```

Comment:
this addition with a bool is hard to translate. Having an `increment` function on `DrlProximity` leads to worse codegen. Instead I've basically written out what the incrementing would do. I think that's clearer, but it is kind of verbose.

Reply:
The translation here looks correct, but I think an `increment` method would have also been fine. Even if codegen isn't optimal, I think we'd rather wait until profiling before worrying too much about the performance cost. But the version you have here is easier to understand in my opinion, so what you have now is fine.